src/slurm_plugin/fleet_manager.py

```diff
@@ -47,12 +47,15 @@ def __hash__(self):
     @staticmethod
     def from_describe_instance_data(instance_info):
         try:
-            return EC2Instance(
-                instance_info["InstanceId"],
-                instance_info["PrivateIpAddress"],
-                instance_info["PrivateDnsName"].split(".")[0],
-                instance_info["LaunchTime"],
-            )
+            for network_interface in instance_info["NetworkInterfaces"]:
+                attachment = network_interface["Attachment"]
+                if attachment["DeviceIndex"] == 0 and attachment["NetworkCardIndex"] == 0:
+                    return EC2Instance(
+                        instance_info["InstanceId"],
+                        network_interface["PrivateIpAddress"],
+                        instance_info["PrivateDnsName"].split(".")[0],
+                        instance_info["LaunchTime"],
+                    )
```

Comment:
**Suggestion:** Can you refactor the code in order to address CodeQL issue.
An approach could be assign `instance_info["PrivateIpAddress"]` as a starting value to a variable. In the loop try to assign the right `network_interface["PrivateIpAddress"]` to that variable. Then use the variable to set the second field of the `return EC2Instance`.
In this way  there is the fallback of `instance_info["PrivateIpAddress"]` and a single explicit return.